### ISSUE TITLE
[dev.icinga.com #13885] Import source for Director

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2-module-businessprocess#69]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/13885

**Created by TheFlyingCorpse on 2017-01-04 22:33:34 +00:00**

Assignee: _(none)_
Status: _New_ 
Target Version: _(none)_
Last Update: _2017-01-04 22:33:34 +00:00 (in Redmine)_

---

Any plans to make BusinessProcess a source for Director so I can auto add them?
- With docu on correct links inside IcingaWeb2, seeing as objects have special URL's depending on where in the hierarchy they are :)

(Maybe this is after the storage method is changed to be in the database, [\#8585](https://dev.icinga.com/issues/8585) ?)

**Comment by Thomas-Gelf** [comment]
A related prototype already exists, it's however too late for v2.0.0.

**Comment by bmtwl** [comment]
9 years later, is there any sign that this feature will reach the main branch? It's my number 1 pain point with this module.
Manually maintaining this for complex systems with large numbers of nodes which come and go is not sustainable.

**Comment by nilmerg** [comment]
No, not any time soon I'm afraid. But it's not hard to implement, the basics ([ITL checkcommand](https://icinga.com/docs/icinga-2/latest/doc/10-icinga-template-library/#business-process), [detail view integration](https://icinga.com/docs/icinga-business-process-modeling/latest/doc/10-Monitoring/)) are there, contributions are welcome, it's just not on our road map right now.

**Comment by bmtwl** [comment]
Has there ever been an attempt to allow definition by server/host group or variable matching so that objects could be expanded dynamically (even if its only during startup)? 
I think that would bridge the gap pretty well without full director integration or a database backend.
